### PR TITLE
Fix mock response class

### PR DESCRIPTION
**Pull request by felixmaier1989**
I'm building an API using CakePHP 2.8 and I wanted to have my own Response class. But if in my controllers I set $_responseClass to something else but `CakeResponse` , in my unit tests, the mocked response object will still mock class CakeResponse.
See https://github.com/cakephp/cakephp/blob/2.x/lib/Cake/TestSuite/ControllerTestCase.php#L279

If I set $_responseClass to `MyCakeResponse` I would have expected the response object to be mocking `MyCakeResponse`.

### CODE REVIEW

File: lib/Cake/Controller/Controller.php [hunk 1]
```diff
@@ -1249,5 +1249,13 @@ protected function _getViewObject() {
 
 		return new $viewClass($this);
 	}
+	
+/**
+ * Returns the class name to use for creating the response object.
+ * @return string
+ */
+	public function getResponseClass() {
+		return $this->_responseCLass;
```
Comment (dereuromark) [inline lib/Cake/Controller/Controller.php]: Typo in Class.

Comment (felixmaier1989) [inline lib/Cake/Controller/Controller.php]: Thank you Mark

File: lib/Cake/Controller/Controller.php [hunk 2]
```diff
@@ -1249,5 +1249,13 @@ protected function _getViewObject() {
 
 		return new $viewClass($this);
 	}
+	
+/**
+ * Returns the class name to use for creating the response object.
+ * @return string
```
Comment (dereuromark) [inline lib/Cake/Controller/Controller.php]: A newline before return tag would be good.

**Comment by dereuromark** [comment]
Looks good to me.

**Comment by markstory** [comment]
I would rather make `ControllerTestCase` accept a response object as a parameter, or have `ControllerTestCase` have a property that defines the classname that will be used.

Exposing a method on the controller when the controller is not responsible for building the response during normal runtime makes me think that the responsibility is being put in the wrong place.

**Pull request closed without merging**